crates/trigger/src/lib.rs

```diff
@@ -101,34 +102,42 @@ impl<Executor: TriggerExecutor> TriggerExecutorBuilder<Executor> {
-                        // configurations.
-
-                        Arc::new(spin_key_value::DelegatingStoreManager::new(
-                            [(
-                                "default".to_owned(),
-                                Arc::new(spin_key_value_sqlite::KeyValueSqlite::new(
-                                    if let Some(key_value_file) =
-                                        key_value_file_location(component.app)
-                                    {
-                                        spin_key_value_sqlite::DatabaseLocation::Path(
-                                            key_value_file,
-                                        )
-                                    } else {
-                                        spin_key_value_sqlite::DatabaseLocation::InMemory
-                                    },
+                    spin_key_value::KeyValueComponent::new(spin_key_value::manager(
+                        move |component| {
+                            // TODO: Once we have runtime configuration for key-value stores, the user will be able
+                            // to both change the default store configuration (e.g. use Redis, or an SQLite
+                            // in-memory database, or use a different path) and add other named stores with their
+                            // own configurations.
+
+                            let init = || {
+                                Arc::new(spin_key_value::DelegatingStoreManager::new(
+                                    [(
+                                        "default".to_owned(),
+                                        Arc::new(spin_key_value_sqlite::KeyValueSqlite::new(
+                                            if let Some(key_value_file) =
+                                                key_value_file_location(component.app)
+                                            {
+                                                spin_key_value_sqlite::DatabaseLocation::Path(
+                                                    key_value_file,
+                                                )
+                                            } else {
+                                                spin_key_value_sqlite::DatabaseLocation::InMemory
+                                            },
+                                        ))
+                                            as Arc<dyn spin_key_value::StoreManager>,
+                                    )]
+                                    .into_iter()
+                                    .collect(),
                                 ))
-                                    as Arc<dyn spin_key_value::StoreManager>,
-                            )]
-                            .into_iter()
-                            .collect(),
-                        ))
-                    })),
+                            };
+
+                            manager.get_or_init(init).clone()
```

Comment:
Can this be done eagerly outside of `KeyValueComponent::new`? Is it slow?

Reply:
I needs `component.app`, which it doesn't have until the callback is called.  Technically, we could rearrange the code in the outer scope to load the app earlier, which is what I was doing earlier, but that was awkward for lifetime reasons (multiple mutable references to the loader, etc.).